### PR TITLE
#1667 Fix k8s informer after upgrades

### LINKED ISSUE #1667: Ingress Events Not Detected After Upgrade of KubernetesClient to v6
### Describe the bug
Kubernetes Ingress events don't get picked up after upgrade the KubernetesClient library as part of #1664 . 
When the ingress controller starts, it does load the ingresses correctly, however subsequent updates to ingresses do not.

### To Reproduce
1. `kubectl apply -f .\samples\KuberenetesIngress.Sample\backend\backend.yaml`
1. `kubectl apply -f .\samples\KuberenetesIngress.Sample\backend\ingress-sample.yaml`
1. `docker build -t yarp-controller:latest -f .\src\Kubernetes.Controller\Dockerfile .`
1. `kubectl apply -f .\samples\KuberenetesIngress.Sample\Ingress\ingress-controller.yaml`
1. `docker build -t backend:latest -f .\samples\KuberenetesIngress.Sample\backend\Dockerfile .`
1. `Invoke-WebRequest http://localhost` fails to be forwarded to the backend resource

Revert the KubernetesClient library to `5.0.20` and see that the `iwr` request works as expected

### PR DESCRIPTION
**Pull request by dpbevin**
* Updated Dockerfiles based on SDK change in `global.json`
* Fixed the "combined" Dockerfile
* Fixed an issue with the annotations for the default ingress class
* Fixes https://github.com/microsoft/reverse-proxy/issues/1667 by removing customized ResourceInformer logic (now using out of the box code from the `KubernetesClient` library

Fixes #1667

### CODE REVIEW

File: src/Kubernetes.Controller/OperatorFramework/KubernetesControllerExtensions.cs [hunk 1]
```diff
@@ -1,36 +1,31 @@
...
 
     /// <summary>
     /// Registers the resource informer.
     /// </summary>
     /// <typeparam name="TResource">The type of the t related resource.</typeparam>
+    /// <typeparam name="TService">The implementation type of the resource informer.</typeparam>
     /// <param name="services">The services.</param>
     /// <returns>IServiceCollection.</returns>
-    public static IServiceCollection RegisterResourceInformer<TResource>(this IServiceCollection services)
+    public static IServiceCollection RegisterResourceInformer<TResource, TService>(this IServiceCollection services)
```
Comment (MihaZupan) [inline src/Kubernetes.Controller/OperatorFramework/KubernetesControllerExtensions.cs:24]: Should this have a `where TService : IResourceInformer<TResource>` constraint?

File: samples/KuberenetesIngress.Sample/Combined/Dockerfile [hunk 2]
```diff
@@ -5,11 +5,11 @@ WORKDIR /app
 EXPOSE 80
 EXPOSE 443
 
-FROM mcr.microsoft.com/dotnet/sdk:6.0.100 AS publish
+FROM mcr.microsoft.com/dotnet/sdk:7.0.100-preview.3 AS publish
```
Comment (MihaZupan) [inline samples/KuberenetesIngress.Sample/Combined/Dockerfile:8]: These should still stay on 6, unless there's a reason we need 7 I'm missing?

File: src/Kubernetes.Controller/OperatorFramework/Informers/ResourceInformer.cs [hunk 3]
```diff
@@ -188,6 +190,8 @@ public override async Task RunAsync(CancellationToken cancellationToken)
             }
         }
 
+        protected abstract Task<Rest.HttpOperationResponse<TListResource>> RetrieveResourceList(bool? watch = null, string resourceVersion = null, CancellationToken cancellationToken = default);
```
Comment (MihaZupan) [inline src/Kubernetes.Controller/OperatorFramework/Informers/ResourceInformer.cs:193]:
```suggestion
        protected abstract Task<Rest.HttpOperationResponse<TListResource>> RetrieveResourceListAsync(bool? watch = null, string resourceVersion = null, CancellationToken cancellationToken = default);
```

File: samples/KuberenetesIngress.Sample/Combined/Dockerfile (hunk 2 above)
Comment (dpbevin) [reply samples/KuberenetesIngress.Sample/Combined/Dockerfile:8]: The SDK version in the `global.json` file got bumped to 7.x by someone else. Without this change the `docker build` command fails.
Note that the runtime for the Docker images remains on 6 (see line 1 of the Dockerfiles).

File: src/Kubernetes.Controller/OperatorFramework/Informers/ResourceInformer.cs (hunk 3 above)
Comment (dpbevin) [reply src/Kubernetes.Controller/OperatorFramework/Informers/ResourceInformer.cs:193]: Will fix

File: src/Kubernetes.Controller/OperatorFramework/KubernetesControllerExtensions.cs (hunk 1 above)
Comment (dpbevin) [reply src/Kubernetes.Controller/OperatorFramework/KubernetesControllerExtensions.cs:24]: Yes, it should. Got spot.

File: samples/KuberenetesIngress.Sample/Combined/Dockerfile (hunk 2 above)
Comment (dpbevin) [reply samples/KuberenetesIngress.Sample/Combined/Dockerfile:8]: As a separate PR, I'll see if I can set up a simple workflow to build the Dockerfiles. I'd probably look at something similar to the workflow used for the docs.

Comment (MihaZupan) [reply samples/KuberenetesIngress.Sample/Combined/Dockerfile:8]: Oh I see, if it's just the SDK that's fine.

**Pull request merged**